client/scripts/stores/RecentListingStore.ts

```diff
@@ -0,0 +1,206 @@
+import moment from 'moment';
+import IdStore from './IdStore';
+import { OffchainThread } from '../models';
+import { orderDiscussionsbyLastComment } from '../views/pages/discussions/helpers';
+
+interface IListingParams {
+  topicName?: string;
+  stageName?: string;
+  pinned?: boolean;
+}
+
+interface IListingFetchState {
+  allThreadsInitialized: boolean;
+  topicInitialized: { [topicName: string]: boolean };
+  stageInitialized: { [stageName: string]: boolean };
+
+  allThreadsCutoffDate: moment.Moment;
+  topicCutoffDate: { [topicName: string]: moment.Moment };
+  stageCutoffDate: { [stageName: string]: moment.Moment };
+
+  allThreadsDepleted: boolean;
+  topicDepleted: { [topicName: string]: boolean };
+  stageDepleted: { [stageName: string]: boolean };
+}
+
+class RecentListingStore extends IdStore<OffchainThread> {
+  private _threads = new Array<OffchainThread>();
+  public get threads() {
+    return this._threads;
+  }
+
+  private _fetchState: IListingFetchState = {
+    allThreadsInitialized: false,
+    topicInitialized: {},
+    stageInitialized: {},
+
+    allThreadsCutoffDate: moment(),
+    topicCutoffDate: {},
+    stageCutoffDate: {},
+
+    allThreadsDepleted: false,
+    topicDepleted: {},
+    stageDepleted: {},
+  };
+
+  public add(thread: OffchainThread) {
+    const existingThread = this.getById(thread.id);
+    if (existingThread) this.remove(existingThread);
+
+    super.add(thread);
+    this._threads.push(thread);
+    return this;
+  }
+
+  public remove(thread: OffchainThread) {
+    const existingThread = this.getById(thread.id);
+    if (!existingThread) return;
+    const proposalIndex = this._threads.indexOf(existingThread);
+    if (proposalIndex === -1) return;
+
+    super.remove(thread);
+    this._threads.splice(proposalIndex, 1);
+    return this;
+  }
+
+  public clear() {
+    super.clear();
+    this._threads = [];
+  }
+
+  // Getters
+
+  public getById(id) {
```

Comment:
Is there any reason we need to declare an override for `getById` + maintain `this._threads`, given that the `IdStore` parent already effectively contains these features? Alternatively we can replace the `IdStore` with a regular `Store` since these features are going unused.